src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -32322,15 +32322,9 @@ public static void Main(string[] args)
                 // (6,33): error CS8081: Expression does not have a name.
                 //         const bool a = M(nameof(M(out int z1)), z1);
                 Diagnostic(ErrorCode.ERR_ExpressionHasNoName, "M(out int z1)").WithLocation(6, 33),
-                // (6,24): error CS0133: The expression being assigned to 'a' must be constant
-                //         const bool a = M(nameof(M(out int z1)), z1);
-                Diagnostic(ErrorCode.ERR_NotConstantExpression, "M(nameof(M(out int z1)), z1)").WithArguments("a").WithLocation(6, 24),
                 // (7,33): error CS8081: Expression does not have a name.
                 //         const bool b = M(nameof(M(out var z2)), z2);
                 Diagnostic(ErrorCode.ERR_ExpressionHasNoName, "M(out var z2)").WithLocation(7, 33),
-                // (7,24): error CS0133: The expression being assigned to 'b' must be constant
-                //         const bool b = M(nameof(M(out var z2)), z2);
-                Diagnostic(ErrorCode.ERR_NotConstantExpression, "M(nameof(M(out var z2)), z2)").WithArguments("b").WithLocation(7, 24),
```

Comment:
I believe this is caused by [this change](https://github.com/dotnet/roslyn/pull/20729/files?diff=unified#diff-9a6ec4cf1baa197e22982c88ae1045deL893) in `Binder_Statements.cs`. If diagnostics from binding the initializer go into `localDiagnostics` there, `localDiagnostics.HasAnyResolvedErrors()` becomes `true` and so  `GetConstantValueDiagnostics()` isn't called.

Preserving exactly the same behavior would be possible, but I don't think it would be worth it.